File: DB/Backend/database/models/festival.py
"""
Pydantic models for Festival data.
Mirrors Frontend/src/types/festival.ts — FestivalEvent.
"""

from typing import Optional, List
from pydantic import BaseModel


class FestivalResponse(BaseModel):
    """Matches Frontend FestivalEvent interface."""
    id: str
    name: str
    marathiName: Optional[str] = None
    hindiName: Optional[str] = None
    monthIndex: int = 0
    dateOrSeason: str = ""
    upcomingDate: Optional[str] = None
    dayOrTithi: Optional[str] = None
    state: str = ""
    stateId: str = ""
    category: str = ""   # harvest/religious/seasonal/art/new-year
    image: str = ""
    shortDescription: str = ""
    culturalSignificance: str = ""
    traditionalPractices: List[str] = []
    relatedItemSlug: Optional[str] = None

File: DB/Frontend/src/types/festival.ts
export interface FestivalEvent {
  id: string;
  name: string;
  marathiName?: string;
  hindiName?: string;
  monthIndex: number; // 0 for Jan, 1 for Feb ... 11 for Dec
  dateOrSeason: string;
  upcomingDate?: string;
  dayOrTithi?: string;
  state: string;
  stateId: string;
  category: 'harvest' | 'religious' | 'seasonal' | 'art' | 'new-year';
  image: string;
  shortDescription: string;
  culturalSignificance: string;
  traditionalPractices: string[];
  relatedItemSlug?: string;
}
